kolena/_experimental/object_detection/evaluator.py

```diff
@@ -0,0 +1,343 @@
+        assert configuration is not None, "must specify configuration"
+        # compute thresholds to cache values for subsequent steps
+        self.compute_f1_optimal_thresholds(configuration, inferences)
+        labels = {gt.label for _, gts, _ in inferences for gt in gts.bboxes}
+        return [(ts, self.compute_image_metrics(gt, inf, configuration, labels)) for ts, gt, inf in inferences]
+
+    def compute_aggregate_label_metrics(
+        self,
+        matchings: List[MulticlassInferenceMatches],
+        label: str,
+        thresholds: Dict[str, float],
+    ) -> ClassMetricsPerTestCase:
+        m_matched = []
+        m_unmatched_gt = []
+        m_unmatched_inf = []
+        tpr_counter = 0
+        fpr_counter = 0
+        confused_counter = 0
+        samples = 0
+        # filter the matching to only consider one class
+        for match in matchings:
+            has_tp = False
+            has_fn = False
+            has_fp = False
+            for gt, inf in match.matched:
+                if gt.label == label:
+                    m_matched.append((gt, inf))
+                    has_tp = True
+            for gt, inf in match.unmatched_gt:
+                if gt.label == label:
+                    m_unmatched_gt.append(gt)
+                    has_fn = True
+                    if inf is None:
+                        confused_counter += 1
+            for inf in match.unmatched_inf:
+                if inf.label == label:
+                    m_unmatched_inf.append(inf)
+                    has_fp = True
+            if has_tp:
+                tpr_counter += 1
+            if has_fp:
+                fpr_counter += 1
+            if has_tp or has_fn or has_fp:
+                samples += 1
+        tp_count = len(m_matched)
+        fp_count = len(m_unmatched_inf)
+        fn_count = len(m_unmatched_gt)
+        precision = tp_count / (tp_count + fp_count) if (tp_count + fp_count) > 0 else 0
+        recall = tp_count / (tp_count + fn_count) if (tp_count + fn_count) > 0 else 0
+        all_bbox_matches = [
+            InferenceMatches(matched=m_matched, unmatched_gt=m_unmatched_gt, unmatched_inf=m_unmatched_inf),
+        ]
+        baseline_pr_plot = compute_pr_plot(all_bbox_matches)
+        baseline_pr_plot = baseline_pr_plot.curves[0] if baseline_pr_plot is not None else None
+        average_precision = (
+            compute_average_precision(baseline_pr_plot.y, baseline_pr_plot.x)
+            if baseline_pr_plot is not None and precision > 0
+            else 0.0
+        )
+
+        return ClassMetricsPerTestCase(
+            Class=label,
+            Threshold=thresholds[label],
+            TestSamples=samples,
+            Objects=tp_count + fn_count,
+            Inferences=tp_count + fp_count,
+            TP=tp_count,
+            FN=fn_count,
+            FP=fp_count,
+            Confused=confused_counter,
+            TPR=tpr_counter / samples if samples > 0 else 0.0,
```

Comment:
Should `TPR` and `FPR` be based on samples or based on objects?

Reply:
I'd consider omitting TPR — not clear to me what it means, and why it's meaningful to include in addition to Recall.

Reply:
Should be objects. I made it samples-based because it is hard to define how many true negatives there are in `FPR = FP / (FP + TN)`. Assuming Gordon wants this removed 👍